internal/pkg/manifest/validate.go

```diff
@@ -653,6 +653,9 @@ func (i ImageWithPort) validate() error {
 // validate returns nil if Image is configured correctly.
 func (i Image) validate() error {
 	var err error
+	if err := i.ImageLocationOrBuild.validate(); err != nil {
+		return err
+	}
 	if err = i.Build.validate(); err != nil {
 		return fmt.Errorf(`validate "build": %w`, err)
```

Comment:
I was just wondering if we should keep this code inside `ImageLocationOrBuild.validate()` as this validate belongs to `ImageLocationOrBuild` and not `Image`?

Reply:
L 659 - 660 can be added to `ImageLocationOrBuild.validate()` and the remaining logic L661-668 where both `build` and `location` are empty should be inside the `Image` (main container) only as this case is not applicable for sidecar containers where both `build` and `location` can be empty

Reply:
Done 
Added these below to `ImageLocationOrBuild` as this case is applicable to both sidecars and main container.
```
if err = i.Build.validate(); err != nil {
    return fmt.Errorf(`validate "build": %w`, err)
}